### PR TITLE
Extend help output with arbitrary content

### PR DESCRIPTION
**Pull request by natemcmaster**
I want to add custom help to dotnet-watch. Adding API to allow users to plugin custom help text.

cc @pranavkm

### CODE REVIEW

**Comment by natemcmaster** [comment]
Ping @cesarbs could you review this minor change? It will allow me to address what we discussed for https://github.com/aspnet/DotNetTools/issues/205

File: test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs [hunk 1]
```diff
@@ -580,5 +580,24 @@ public void HelpTextShowsArgSeparator()
...
+
+        [Fact]
+        public void HelpTextShowsExtendedHelp()
+        {
+            var app = new CommandLineApplication()
+            {
+                Name = "befuddle",
+                ExtendedHelpText = @"
+Remarks:
+  This command is hard to use that I want to include examples in the help text.
```
Comment (cesarbs) [inline test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs]: _so_ hard?

**Comment by cesarbs** [comment]
Can you show me how you'd use that to improve the `dotnet-watch` usage message?

**Comment by natemcmaster** [comment]
```c#
            var app = new CommandLineApplication()
            {
                Name = "dotnet watch",
                ExtendedHelpText = @"
Remarks:
  The argument separator '--' can be used to separate 'dotnet-watch' arguments from arguments 
  passed to the subprocess. For example, 'dotnet watch --quiet -- test -class TestClass1'

Examples:
  dotnet watch run
  dotnet watch test
"
            };
```

Would output:
```
Usage: dotnet watch [options] [[--] <arg>...]

Options:
  -h|-?|--help  Show help information
  -v|--verbose  Verbose output
  -q|--quiet    Suppress most output messages

Remarks:
  The argument separator '--' can be used to separate 'dotnet-watch' arguments from arguments
  passed to the subprocess. For example, 'dotnet watch --quiet -- test -class TestClass1'

Examples:
  dotnet watch run
  dotnet watch test
```

**Pull request merged**